### PR TITLE
Use range algorithms in spec

### PR DESCRIPTION
**Pull request by apasel422**
***
<a href="https://pr-preview.s3.amazonaws.com/apasel422/attribution-reporting-api/pull/1085.html" title="Last updated on Oct 24, 2023, 2:46 PM UTC (86fef5b)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/1085/bc9f1df...apasel422:86fef5b.html" title="Last updated on Oct 24, 2023, 2:46 PM UTC (86fef5b)">Diff</a>

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -1864,8 +1864,8 @@ To <dfn>parse trigger specs</dfn> given a [=map=] |map|, a [=moment=]
     1. Let |spec| be a new [=trigger spec=] with the following items:
         : [=trigger spec/event-level report windows=]
         :: |defaultReportWindows|
-    1. For each integer |triggerData| between 0 (inclusive) and
-        |defaultTriggerDataCardinality| (exclusive):
+    1. [=set/iterate|For each=] integer |triggerData| of [=the exclusive range|the range=] 0 to
```
Comment (linnan-github) [inline index.bs:1867]: What's the benefit of not displaying "the exclusive range" explicitly, but show "exclusive" at the end?

Comment (apasel422) [reply index.bs:1867]: That is what the Infra spec does: https://infra.spec.whatwg.org/#example-the-range

Comment (apasel422) [reply index.bs:1867]: It also makes it more clear that the end of the range is exclusive, rather than the beginning.

Comment (linnan-github) [reply index.bs:1867]: Got it, thanks.

**Pull request merged**